File: php/find_similar_places.php
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <script type="text/javascript" src="http://maps.google.com/maps/api/js?sensor=false"></script>
  <?php

  if (isset($_POST["lat"])) {
    $lat = htmlspecialchars($_POST["lat"]);
  } else {
    $lat = "35.7886";
  }
  if (isset($_POST["long"])) {
    $long = htmlspecialchars($_POST["long"]);
  } else {
    $long = "139.6857";
  }
  ?>
  <script type="text/javascript">
    function initialize() {
      var latlng = new google.maps.LatLng(<?=$lat?>, <?=$long?>);
      var myOptions = {
        zoom: 10,
        center: latlng,
        mapTypeId: google.maps.MapTypeId.ROADMAP
      };
      var map = new google.maps.Map(document.getElementById("map_canvas"), myOptions);
      google.maps.event.addListener(map, 'mouseup', function() {
        // 地図をドラッグしたらフォールの緯度経度欄に反映させる
        document.getElementById("lat").value = map.getCenter().lat();
        document.getElementById("long").value = map.getCenter().lng();
      });
    }

  </script>
</head>
<body onload="initialize()">
  <?php
  // tmhOAuthを初期化
  require "./tmhOAuth.php";
  $twitter = new tmhOAuth(array());

  if (isset($_REQUEST["command"])) {
    if ("similar_places" == $_REQUEST["command"]) {
      // 場所を検索する
      $twitter->request("GET", $twitter->url("1/geo/similar_places")
        , array("lat" => $_POST["lat"]
        , "long" => $_POST["long"]
        , "name" => $_POST["name"]
        ));
      $places = json_decode($twitter->response["response"])->{"result"}->{"places"};

      // ヒットした場所を一覧表示
      print "緯度: $lat 経度: $long 付近 名称: " . htmlspecialchars($_POST["name"]) . " でヒットした場所:<br>";
      foreach ($places as $place) {
        $fullName = $place->{"full_name"};
        print htmlspecialchars($fullName) . "<br>";
      }
    }
    ?><br>
    <?php

  }
  ?>

  <div id="map_canvas" style="width:300px; height:300px"></div>

  <form action="./find_similar_places.php" method="POST">
    <input type="hidden" name="command" value="similar_places"/>
    緯度:<input type="text" name="lat" id="lat"/><br>
    経度:<input type="text" name="long" id="long"/><br>
    名称:<input type="text" name="name"/><br>
    <input type="submit" value="場所を検索"/>
  </form>
</body>
</html>

File: java/src/main/java/FindSimilarPlaces.java
import twitter4j.GeoLocation;
import twitter4j.Place;
import twitter4j.SimilarPlaces;
import twitter4j.Twitter;
import twitter4j.TwitterException;
import twitter4j.TwitterFactory;

public final class FindSimilarPlaces {
  public static void main(String[] args) {
    if (args.length < 3) {
      System.out.println("使用法: java FindSimilarPlaces [緯度] [経度] [名前]");
      System.exit(-1);
    }
    try {
      Twitter twitter = new TwitterFactory().getInstance();

      GeoLocation location = new GeoLocation(Double.parseDouble(args[0]), Double.parseDouble(args[1]));
      String name = args[2];
      SimilarPlaces places = twitter.getSimilarPlaces(location, name, null, null);
      System.out.println("token: " + places.getToken());
      if (places.size() == 0) {
        System.out.println("指定した条件で場所は見つかりませんでした");
      } else {
        for (Place place : places) {
          System.out.println("id: " + place.getId() + " 名前: " + place.getFullName());
        }
      }
      System.exit(0);
    } catch (TwitterException te) {
      te.printStackTrace();
      System.out.println("API呼び出しに失敗しました: " + te.getMessage());
      System.exit(-1);
    }
  }
}
